packages/material-ui/src/Slider/Slider.js

```diff
@@ -405,6 +405,14 @@ const Slider = React.forwardRef(function Slider(props, ref) {
     name: 'Slider',
   });
 
+  const onChange =
+    onChangeProp &&
+    ((event, value) => {
+      event.target.name = name;
```

Comment:
This is actually mutating the underlying DOM element which might be problematic. 
1. It introduces global side-effects
2. It might be slow due to expando properties

Reply:
Do you have a specific issue in mind? What if we were to wait to see if it turn into an issue?

Reply:
Are you suggesting we do this instead?

https://github.com/mui-org/material-ui/blob/27fe6a30e5e5432036e8a0ce8a409b75dfb7f2dc/packages/material-ui/src/Select/SelectInput.js#L185-L190

Reply:
> Do you have a specific issue in mind?

We're mutating elements. That is the issue.

Reply:
Overriding the events target with defineProperty seems to be a good choice. 

Users can always access the native events if needed. Is the performance of event.persist() a concern? 

Let me know what's best, or perhaps it's better not to handle these cases, and I'll make the changes.

Reply:
> Is the performance of event.persist() a concern?

No, it even becomes the default behavior in React 17.

Reply:
It seems like the mousemove event returns a native event. Once we override the target, the original event target will no longer be accessible.

Reply:
I guess it's fine?